va/validation-authority.go

```diff
@@ -447,24 +453,30 @@ func (va *ValidationAuthorityImpl) getCAASet(hostname string) (*CAASet, error) {
 			break
 		}
 
-		// Query CAA records for domain and its alias if it has a CNAME
-		for _, alias := range []bool{false, true} {
-			if alias {
-				target, _, err := va.DNSResolver.LookupCNAME(queryDomain)
-				if err != nil {
-					return nil, err
-				}
-				queryDomain = target
-			}
-			CAAs, _, err := va.DNSResolver.LookupCAA(queryDomain)
+		cnames := 0
+		for {
```

Comment:
According to everyone except the CAA RFC, it's a DNS error to have both CNAME and anything else (except specific DNSSEC RRs) for the same name, so I was actually thinking we should return an error in that case -- but indeed, CAA specifies non-error behavior for that case. Unless you define the result of "performing a query" to be an error if CNAME and CAA both exist... but the way they're deliberately ordered looks more like the RFC just forgot you're not allowed to have both.

This seems to create a small problem that compliant CAA resolution requires upstream DNS caches to pass invalid DNS data through to us reliably. Not sure how/whether to deal with that as a deployment issue, but at least it's (almost completely) clear what _this_ code has to do.

rfc6844 also specifies that DNAME must be followed, which we aren't attempting at all yet.

Conveniently rfc6844 does not specify what to do if both CNAME and DNAME exist for a label, so presumably we should treat _that_ error case as an error.

Will fix.